ingest-app/src/cmr/ingest/data/granule_bulk_update.clj

```diff
@@ -84,22 +103,50 @@
   cmr.oracle.connection.OracleStore
 
   (get-granule-tasks-by-provider-id
-   [db provider-id]
-   (jdbc/with-db-transaction
-    [conn db]
-    ;; Returns a list of bulk update tasks for the provider
-    (let [stmt (sql-utils/build
-                (sql-utils/select
-                 [:created-at :name :task-id :status :status-message :request-json-body]
-                 (sql-utils/from "granule_bulk_update_tasks")
-                 (sql-utils/where `(= :provider-id ~provider-id))
-                 (sql-utils/order-by (sql-utils/desc `(+ :task-id 0)))))
-          ;; Note: the column selected out of the database is created_at, instead of created-at.
-          statuses (doall (map #(update % :created_at (partial oracle/oracle-timestamp->str-time conn))
-                               (sql-utils/query conn stmt)))
-          statuses (map util/map-keys->kebab-case statuses)]
-      (map #(update % :request-json-body util/gzip-blob->string)
-           statuses))))
+   [db provider-id params]
+   (try
+     (jdbc/with-db-transaction
+      [conn db]
+      ;; Returns a list of bulk update tasks for the provider
+      (let [max-rows (config/granule-bulk-update-tasks-max-rows)
+            [begin-date end-date] (get-begin-end-dates (:date params))
+            stmt (if (and begin-date end-date)
+                   (sql-utils/build
+                    (sql-utils/select
+                     [:created-at :name :task-id :status :status-message :request-json-body]
+                     (sql-utils/from "granule_bulk_update_tasks")
+                     (sql-utils/where `(and (= :provider-id ~provider-id)
+                                            (<= :rowNum ~max-rows)
+                                            (<= :created-at (:to_utc_timestamp_tz ~end-date)) 
+                                            (>= :created-at (:to_utc_timestamp_tz ~begin-date))))
+                     (sql-utils/order-by (sql-utils/desc `(+ :task-id 0)))))
+                   (if begin-date
+                     (sql-utils/build
+                      (sql-utils/select
+                       [:created-at :name :task-id :status :status-message :request-json-body]
+                       (sql-utils/from "granule_bulk_update_tasks")
+                       (sql-utils/where `(and (= :provider-id ~provider-id)
+                                              (<= :rowNum ~max-rows)
+                                              (>= :created-at (:to_utc_timestamp_tz ~begin-date))))
+                       (sql-utils/order-by (sql-utils/desc `(+ :task-id 0)))))
+                     (sql-utils/build
```

Comment:
Is this section mean to be for the event where the `endDate` is the only one provided? If so doesn't it need a `(<= :created-at (:to_utc_timestamp_tz ~end-date` SQL clause? Or is that not a valid use-case by itself?

Reply:
No. If you read the api.doc,  you will see that when one date is provided, it's treated as the begin date.